testing.go

```diff
@@ -5,18 +5,39 @@ import (
 	"testing"
 )
 
-// SkipIfProviderIsNotHealthy is a utility function capable of skipping tests
-// if the provider is not healthy, or running at all.
-// This is a function designed to be used in your test, when Docker is not mandatory for CI/CD.
-// In this way tests that depend on Testcontainers won't run if the provider is provisioned correctly.
-func SkipIfProviderIsNotHealthy(t *testing.T) {
-	ctx := context.Background()
+// SkipIfProviderIsNotHealthWithContext is a context configurable varient of
+// SkipIfProviderIsNotHealthy
+func SkipIfProviderIsNotHealthyWithContext(tb testing.TB, ctx context.Context) {
 	provider, err := ProviderDocker.GetProvider()
 	if err != nil {
-		t.Skipf("Docker is not running. TestContainers can't perform is work without it: %s", err)
+		tb.Skipf("Docker is not running. TestContainers can't perform this work without it: %s", err)
 	}
+
 	err = provider.Health(ctx)
 	if err != nil {
-		t.Skipf("Docker is not running. TestContainers can't perform is work without it: %s", err)
+		tb.Skipf("Docker is not running. TestContainers can't perform this work without it: %s", err)
 	}
 }
+
+// SkipIfProviderIsNotHealthy is a utility function capable of skipping tests
+// if the provider is not healthy, or running at all.
+// This is a function designed to be used in your test, when Docker is not mandatory for CI/CD.
+// In this way tests that depend on Testcontainers won't run if the provider is provisioned correctly.
+func SkipIfProviderIsNotHealthy(tb testing.TB) {
+	SkipIfProviderIsNotHealthyWithContext(tb, context.Background())
+}
+
+// CleanupContainer terminates the passed container after testing.TB has
+// ended.
+func CleanupContainer(tb testing.TB, ctx context.Context, ctr Container) {
```

Comment:
@kiview @eddumelendez @HofmeisterAn wdyt about this helper function? I see it handy as a test resource to be used by consumers of the library. Do you have a similar one in other languages?

Reply:
This looks like it is similar to the `testcontainers-java` `stop()` method. It is defined through the `Startable` interface in `testcontainers-java`:
https://github.com/testcontainers/testcontainers-java/blob/08f75a5217f601439ef426b059908164a02deba4/core/src/main/java/org/testcontainers/lifecycle/Startable.java

So does this try to replicate the automatic `close()` behavior from Java?

From the PR description, it is unclear to me, why `CleanupContainer` was added in this PR.

Reply:
The Container struct has already a `Terminate` function, that would "mimic" Java's stop method. To replicate java's Closeable interface, Go has the `defer` keyword.

At the same time, there is a `terminateContainerOnEnd(tb testing.TB, ctx context.Context, ctr Container)` function, which basically does the same. Probably worth it to move it from docker_test.go to testing.go, better in a separate PR, removing  it from this PR.

@hhsnopek wdyt?

Reply:
It definitely can be removed from this PR 👍

I can open a secondary PR with the exposure of it as a feature as @mdelapenya suggested. I'll refactor this PR with the required changes this week.

Reply:
resolved in latest push